Wip/emnlp autoqa camera ready

rayslxu opened the pull request.

gcampax commented on tool/autoqa/schemaorg/manual-annotations.js:
```diff
@@ -183,6 +183,10 @@ const PROPERTY_CANONICAL_OVERRIDE = {
     },
+    'postalCode': {
+        base: ['postal code', 'postcode', 'zip code'],
+        preposition: ['in #', 'from #'],
```
`in the # zip code` perhaps?

gcampax commented on tool/autoqa/lib/canonical-extractor.js:
```diff
@@ -27,10 +27,13 @@ const child_process = require('child_process');
+    "4": ["four"],
+    "feb 14 2017": ["february 14 2017", "february 14, 2017", "feb 14, 2017", "14 february 2017", "14 february, 2017", "14 feb 2017", "14 feb, 2017"],
+    "may 4th, 2016": ["may 4 2016", "may 4, 2016", "may 4th 2016", "4th may 2016", "4th may, 2016", "4 may 2016", "4 may, 2016"],
+    "august 2nd 2017": ["august 2, 2017", "august 2 2017", "august 2nd, 2017", "2 august 2017", "2nd august 2017", "2 august, 2017", "2nd august, 2017"]
```
Is it worth using the tokenizer to extract these? We can already identify dates pretty well.

rayslxu replied:
Yeah, the extractor will be updated soon with an actual parser. I will do a separate PR to clean this up altogether.

gcampax commented on tool/autoqa/schemaorg/process-schema.js:
```diff
@@ -318,6 +318,8 @@ class SchemaProcessor {
                 metadata.counted_object = MANUAL_COUNTED_OBJECT_OVERRIDE[propertyname];
             else if (propertyname.startsWith('numberOf'))
                 metadata.counted_object = [ clean(propertyname.slice('numberOf'.length)) ];
+            else if (propertyname.startsWith('num') && propertyname.charAt(3) === propertyname.charAt(3).toUpperCase())
```
Writing this as a regular expression would be a lot readable...

gcampax commented on tool/autoqa/schemaorg/manual-annotations.js:
```diff
@@ -351,6 +351,10 @@ const MANUAL_PROPERTY_CANONICAL_OVERRIDE = {
         adjective: ['# speaking']
     },
+    addressLocality: {
+        base: ['city'],
```
city, town, area, etc.

gcampax commented on tool/autoqa/lib/bert-canonical-annotator.py:
```diff
@@ -212,6 +213,10 @@ def predict_one(self, table, arg, query, word, k):
         :return: a array in length k of predicted tokens
         """
+        # skip uncommon word
+        if not wordnet.synsets(word):
```
Is it worth to bring in all of nltk and the wordnet dataset just to learn that a word is common or not? Seems like an overkill...

rayslxu replied:
I searched around and didn't see a better solution... Any suggestion?

gcampax replied:
We make our own vocabulary file (perhaps with wordnet as the starting point) and just load that. A single vocabulary file will be a lot smaller than nltk + wordnet.

gcampax commented on tool/autoqa/auto-annotate.js:
```diff
@@ -68,19 +68,14 @@ module.exports = {
+        parser.add_argument('--algorithms', {
+            nargs: '*',
+            help: 'Different algorithms to generate canonicals including bert-property-synonyms, bart-paraphrase, bert-adjectives, bert-domain-synonyms'
         });
```
What's the default here? Should we have a reasonable default? Is that BART only?

rayslxu replied:
The current default is nothing - we skip auto annotation, but yeah, I think we should set it to the one that we think is the best

gcampax commented on starter/schemaorg/Makefile:
```diff
@@ -91,13 +99,13 @@ custom_train_nlu_flags ?=
+models/paraphraser-bart-large-speedup-megabatch-5m:
+	mkdir -p models
+	wget --no-verbose https://almond-static.stanford.edu/test-data/paraphraser-bart-large-speedup-megabatch-5m.tar.xz
+	tar -C models -xvf paraphraser-bart-large-speedup-megabatch-5m.tar.xz
```
Do we need to distribute both paraphraser models? Can we kill the old one?

rayslxu replied:
you can kill the other one

gcampax replied:
What i mean is overwrite the old model and keep the old URL, but point to the new model.

gcampax reviewed (commented):
Most of the changes are in autoqa's canonical generation, so I only gave a cursory look, cause I'm not very familiar with the code. If you want I can read more carefully, but I trust your tests really. I left a few comments here and there.

rayslxu commented:
@gcampax I addressed most of the comments. 
For default options of algorithms, we will need a path to the paraphraser model if we want to set bart as the default, so I leave baseline as the default (skip auto annotate)
For the paraphraser model, I would rather keep the name.
